File: geeksforgeeks/top10algorithms/string:array/reverse_string.go
// Reverse an array without affecting special characters
// Time Complexity:  O(n)
package main

import ("fmt"
  "unicode")

func main () {
  var str string = "a!!!b.c.d,e'f,ghi"
  l := 0;
  r := len(str)-1;

  fmt.Printf("Length:%d",r)

  for l < r {

    if(!unicode.IsSymbol(str[l])) {
      l++;
    }else if(!unicode.IsSymbol(str[r])) {
      r--;
    }else {
      temp:=str[l];
      str[l] = str[r];
      str[r] = temp;
      l++;
      r--;
    }
  }
}

File: geeksforgeeks/top10algorithms/string:array/reverse_string.cpp
// Reverse an array without affecting special characters
// Time Complexity:  O(n)
#include<iostream>
#include<stdlib.h>
#include<string.h>
using namespace std;


bool isChar (char c) {
  return (c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z');
    //return true;

}
int main() {
  char str[] = "a!!!b.c.d,e'f,ghi";
  int l=0,r=strlen(str) - 1;

  while(l < r) {
    if(!isChar(str[l])) {
      l++;
    }else if(!isChar(str[r])){
      r--;
    }else {
      swap(str[l],str[r]);
      l++;
      r--;
    }
  }

  cout<<"\nAfter reversing string:"<<str<<endl;
  return 0;
}
